src/cli.ts

```diff
@@ -174,7 +179,10 @@ async function prepareSuites(inputs: string[]) {
   /**
    * Validate and handle configs
    */
-  const config = readConfig(environment, options.config);
+  const config =
+    options.config || !options.inline
```

Comment:
This is changing the behavior of our current runner, by default synthetics looks for config files recursively till root directory https://github.com/lucasfcosta/synthetics/blob/f9d4eba2cc337175b6e97137b1d5cd7349b4f863/src/config.ts#L45-L47

Reply:
Hey @vigneshshanmugam, indeed, this does change the default current behaviour of the runner. That's because when running inline we would use the `cwd` to find the configuration.

That can cause a few problems which, in my view, are very significant:
* IMO this design doesn't comply with the "least surprising behaviour" principle.
    Configuration would be read from `cwd`, not the files directory, which could be confusing for users as they could simply run synthetics from another directory and thus have totally different configurations applied.
* When running tests inline, you're piping a journey's contents into the executable for testing. Therefore, the options you'd usually use for suites/directories wouldn't apply most of the time (you can still pass a configuration file path though). This is the same behaviour as Jest, and thus is more compliant with the JS testing environment "standards" (or at least their usual behaviour).

Btw that's as per the AC in https://github.com/elastic/synthetics/issues/407.

Do you think we should change it? I'm sorry but I didn't understand exactly whether you suggested a change here, and, if so, what that change is.

Reply:
Apologies for the lack of detail here. I agree with you on the inline journeys here, we dont need to use config unless specified. 

I was going for this. Assume this is the folder structure
```
 - example/todos/
    - a.journeys.ts
    - b.journeys.ts
    - synthetics.config.ts
    - package.json
```

Previous to this PR, when the runner is invoked within `example/todos`, it picks up the config from `synthetics.config.ts`.

With the current PR, the runner does not run with the config. Is this intended change?

Reply:
Thanks for taking the time to explain 💖 

In the case you mentioned, the configuration _will_ be picked up, though. The configuration will only be ignored if you're using the inline flag, meaning you're piping a script into the runner's `stdin`.

This line:

```js
const config = options.config || !options.inline
      ? readConfig(environment, options.config)
      : {};
```

Will ensure that if you're _not_ running an inline script, `config` will be truthy and thus it will be read from the passed directory.

I've just tested this exact case by doing the following:

1. I created a new directory with two files: a journey, and a configuration file. The configuration file in that folder will write to the console when it's loaded. For the sake of this example, let's imagine these files' folder is `/synth-example`.
    ```js
    // example.journey.ts
    const { journey, step } = require('@elastic/synthetics');

    journey('my journey', ({ page }) => {
      step('google', async () => {
        await page.goto('https://www.google.com');
      });
    });
    ```
    
    ```js
    // synthetics.config.js
    module.exports = (() => {
      console.log('CONFIG LOADED');
      return {};
    })();
    ```
2. I checked out this branch and built the runner with `npm run build`
3. I then `cd`ed into `/synthe-example` and invoked the built runner using the built `cli` for the package `/path_to_synthetics_package/synthetics/dist/cli.js .`. As you can see below, the config was loaded.
    ```
    $ ../dist/cli.js .
    CONFIG LOADED

    Journey: my journey
       ✓  Step: 'google' succeeded (726 ms)

     1 passed (2373 ms)
    ```
4. When piping the contents of `example.journey.js` into the built `cli.js`, I don't see the `CONFIG LOADED` string written to `stdout`, as configuration wasn't loaded.
    > Please notice that you need to convert your journey to an inline journey to run it inline (that means using only the globally provided `step`).
    ```js
    // example-inline.journey.js
    step('google', async () => {
      await page.goto('https://www.google.com');
    });
    ```
    Results:
    ```
    ➜ cat example-inline.journey.js | ../dist/cli.js --inline
    Journey: inline
       ✓  Step: 'google' succeeded (1155 ms)
     1 passed (2541 ms)
    ```


@vigneshshanmugam if you're happy with the explanation above just lmk and I can go ahead and merge. If you think we should change this behaviour please let me know, but I do want to reassure you that in the case you mentioned, the configuration will be read from the passed directory. It's only _not_ read when using the `--inline` flag.

Reply:
My bad, I somehow thought we have changed the behaviour of config loading for suites. Thanks for the detailed explanation and apologies for not looking through the code correctly. PR looks good.